tests/AerospikeCacheTest.php

```diff
@@ -3,7 +3,200 @@
+     */
+    public function testShouldDeleteItem(int $aerospikeStatusCode, bool $expectedDeleteToSucced): void
+    {
+        $aerospikeMock = $this->createMock(\Aerospike::class);
+        $aerospikeCache = new AerospikeCache($aerospikeMock, 'test', 'cache', 'testNamespace');
+
+        $aerospikeMock->method('initKey')
+            ->willReturn(['foo']);
+        $aerospikeMock->method('remove')
+            ->willReturn($aerospikeStatusCode);
+
+        $deleteSuccessful = $aerospikeCache->deleteItem('foo');
+
+        $this->assertSame($expectedDeleteToSucced, $deleteSuccessful);
+    }
+
+    public function provideStatusCodes(): array
+    {
+        return [
+            [\Aerospike::OK, true],
+            [\Aerospike::ERR_CLIENT, false],
+        ];
+    }
+
+    /**
+     * @dataProvider doFetchProvider
+     */
+    public function testShouldReadExistingRecordsFromAerospike(array $cacheItemKeys, array $valuesReturnedByAerospike): void
+    {
+        $mockedAerospikeKeys = array_map(
+            function ($key) {
+                return ['ns' => 'aerospike', 'set' => 'cache', 'key' => $key];
+            },
+            $cacheItemKeys
+        );
+
+        $aerospikeMock = $this->createMock(\Aerospike::class);
+        $aerospikeCache = new AerospikeCache($aerospikeMock, 'test', 'cache');
+
+        $aerospikeMock
+            ->method('initKey')
+            ->willReturnOnConsecutiveCalls(...$mockedAerospikeKeys);
+
+        $aerospikeMock->expects($this->once())
+            ->method('getMany')
+            ->with($this->equalTo($mockedAerospikeKeys))
+            ->willReturnCallback(function ($keys, &$records) use ($valuesReturnedByAerospike) {
+                foreach ($keys as $key) {
+                    if (isset($valuesReturnedByAerospike[$key['key']])) {
+                        $record = [
+                            'key' => $key,
+                            'bins' => ['data' => $valuesReturnedByAerospike[$key['key']]],
+                            'metadata' => ['ttl' => 1000, 'generation' => 2],
+                        ];
+                    } else {
+                        $record = [
+                            'key' => $key,
+                            'bins' => null,
+                            'metadata' => null,
+                        ];
+                    }
+
+                    $records[] = $record;
+                }
+
+                return \Aerospike::OK;
+            });
+
+        $items = $aerospikeCache->getItems($cacheItemKeys);
+
+        foreach ($items as $item) {
+            if (isset($valuesReturnedByAerospike[$item->getKey()])) {
+                $this->assertTrue($item->isHit());
+                $this->assertEquals($valuesReturnedByAerospike[$item->getKey()], $item->get());
+            } else {
+                $this->assertFalse($item->isHit());
+            }
+        }
+    }
 
+    public function doFetchProvider(): array
+    {
+        return [
```

Comment:
Could you provide a name for those cases? Because I don't see what are they for.

Provider also has a wrong name with suffix.

Reply:
(Although it is best practice, names of datasets are also just "nice to have" IMHO, I'd focus on moving the PR forward and possibly add these things later.)

But the provider name should be definitely unified.

Reply:
Yes, it's "_nice to have_" but without it, I can't understand the test...